src/elfcore.c

```diff
@@ -1750,6 +1779,28 @@ int InternalGetCoreDump(void *frame, int num_threads, pid_t *pids,
       SET_FRAME(*(Frame *)frame, thread_regs[i]);
     }
     hasSSE = 0;
+    #elif defined(__aarch64__)
+    memset(scratch, 0xFF, sizeof(scratch));
+    struct iovec scratch_iovec = { scratch, sizeof(scratch)};
+    if (sys_ptrace(PTRACE_GETREGSET, pids[i], (void*)NT_PRSTATUS, &scratch_iovec) == 0) {
+      memcpy(thread_regs + i, scratch, sizeof(struct regs));
+      if (main_pid == pids[i]) {
+        SET_FRAME(*(Frame *)frame, thread_regs[i]);
+      }
+      memset(scratch, 0xFF, sizeof(scratch));
+      scratch_iovec.iov_len = sizeof(scratch);
+      if (sys_ptrace(PTRACE_GETREGSET, pids[i], (void*)NT_FPREGSET, &scratch_iovec) == 0) {
```

Comment:
ptrace() defines the return code to be -1 on error.  Is it better to check for >=0 here?  Or invert the test and check for <0 and use the goto in the then-block, then you don't need the else statement at all.